src/libraries/Common/src/System/Threading/AsyncOverSyncWithIoCancellation.cs

```diff
@@ -65,27 +51,24 @@ public void UnsafeOnCompleted(Action continuation)
         [AsyncMethodBuilder(typeof(PoolingAsyncValueTaskMethodBuilder))]
         public static async ValueTask InvokeAsync<TState>(Action<TState> action, TState state, CancellationToken cancellationToken)
         {
-            // Create the work item state object.  This is used to pass around state through various APIs,
-            // while also serving double duty as the work item used to queue the operation to the thread pool.
-            var workItem = new AsyncOverSyncWithIoCancellation();
-
-            // Queue the work to the thread pool.  This is implemented as a custom awaiter that queues the
-            // awaiter itself to the thread pool.
-            await workItem;
+            // Queue the work to complete asynchronously.
+            await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
```

Comment:
Is there a difference between this and `await Task.Yield().ConfigureAwait(false)` / `await Task.Yield().ConfigureAwait(ConfigureAwaitOptions.None)` ?